src/iface/ethernet.rs

```diff
@@ -362,11 +365,14 @@ impl<'a, 'b, 'c, DeviceT: Device + 'a> Interface<'a, 'b, 'c, DeviceT> {
             _ if handled_by_raw_socket =>
                 Ok(Packet::None),
 
-            _ => {
+            _ if eth_frame.dst_addr().is_unicast() && ipv4_repr.dst_addr.is_unicast() => {
+                // If the datagram does not have a payload greater than or equal
+                // to 64 bits, only send back the size of given payload.
+                let payload_len = min(ip_payload.len(), SIZEOF_ICMP_ERR_PAYLOAD);
```

Comment:
Just push the entire payload there, up to MTU. It's more useful when debugging networking issues anyway, and it's free for us because we don't allocate.

Reply:
Updated.